src/oas3/service.ts

```diff
@@ -67,13 +69,28 @@ export const transformOas3Service: Oas3HttpServiceTransformer = ({ document }) =
       if (!sec) return null;
 
       return keys(sec).map(key => {
-        return securitySchemes.find(securityScheme => {
-          return securityScheme.key === key;
-        });
+        const ss = securitySchemes.find(securityScheme => securityScheme.key === key);
+        if (ss && ss.type === 'oauth2') {
+          return {
+            ...ss,
+            flows: mapValues(ss.flows, (flow: OAuthFlowObject) => ({
+              ...flow,
+              scopes: pickBy(flow.scopes, (_val: string, scopeKey: string) => {
+                const secKey = sec[key];
+                if (secKey) return secKey.includes(scopeKey);
+                return false;
+              }),
+            })),
+          };
+        }
+
+        return ss;
       });
     }),
   );
+
   if (security.length) {
+    //@ts-ignore I hate doing this, but unfortunately Lodash types are (rightfully) very loose and put undefined when it can't happen
```

Comment:
Instead of ts-ignore, why not ditch the lodash mapValues and use a for in loop? I tried it locally and it removes the typing issue

```ts
          const flows = {};
          for (const key in ss.flows) {
            const flow = ss.flows[key];
            flows[key] = {
              ...flow,
              scopes: pickBy(flow.scopes, (_val: string, scopeKey: string) => {
                const secKey = sec[key];
                if (secKey) return secKey.includes(scopeKey);
                return undefined;
              }),
            };
          }
```

Reply:
That is a great idea. I'll do the change now.